tapdb/sqlite.go

```diff
@@ -149,6 +157,87 @@ func NewSqliteStore(cfg *SqliteConfig) (*SqliteStore, error) {
 	return s, nil
 }
 
+// backupSqliteDatabase creates a backup of the given SQLite database.
+func backupSqliteDatabase(srcDB *sql.DB, dbFullFilePath string) error {
+	if srcDB == nil {
+		return fmt.Errorf("backup source database is nil")
+	}
+
+	// Create a database backup file full path from the given source
+	// database full file path.
+	//
+	// Get the current time and format it as a Unix timestamp in
+	// nanoseconds.
+	timestamp := time.Now().UnixNano()
+
+	// Add the timestamp to the backup name.
+	backupFullFilePath := fmt.Sprintf(
```

Comment:
From https://sqlite.org/lang_vacuum.html

'The file named by the INTO clause must not previously exist, or else it must be an empty file, or the VACUUM INTO command will fail with an error.'

So we should check that the path isn't non-empty before we try the backup, and make an empty file there before the actual VACUUM.

Reply:
I read that as: if a file does exist, it must be empty, otherwise it will fail with an error.

If a file does not exist at the backup path at all, then VACUUM INTO will just create one. I don't think we need to add an empty file as a sort of placeholder. And I think we can skip checking for an existing file because of the nanosecond timestamp included in the file name.

Reply:
I read it the same way, should have tagged it with a nit. As in, the timestamp should indeed protect us from a file conflict.

Was suggesting the placeholder as a way to catch some errors before we call VACUUM vs. have SQLite err and possibly return less useful information on err.

Non-blocking.

Reply:
> Was suggesting the placeholder as a way to catch some errors before we call VACUUM vs. have SQLite err and possibly return less useful information on err.

I'm not sure what error info that we might catch with a placeholder that we wouldn't have otherwise seen. Do you know of any? I appreciate that this is a nit now. Just curious.